### PR TITLE
RavenDB-13920 - When out of disk space, don't attempt to log

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-13920

### CODE REVIEW

File: src/Sparrow/Logging/LoggingSource.cs [hunk 1]
```diff
@@ -659,24 +661,14 @@ private void BackgroundLogger()
                     {
                         Console.Error.WriteLine("ERROR! Out of memory exception while trying to log, will avoid logging for the next 5 seconds");
 
-                        var time = 5000;
-                        var current = Stopwatch.GetTimestamp();
+                        DisableLogsFor(threadStates, TimeSpan.FromSeconds(5));
+                    }
+                    catch (IOException ioe) when (IsOutOfDiskSpaceException(ioe))
```
Comment (ayende) [inline src/Sparrow/Logging/LoggingSource.cs]: Why is this just for out of disk space?
Shouldn't we do that for pretty much everything? (No permission on drive, for example)

File: src/Sparrow/Logging/LoggingSource.cs [hunk 2]
```diff
@@ -688,11 +680,51 @@ private void BackgroundLogger()
...
+                {
+                    DiscardThreadLogState(threadStateRef);
+                }
+
+                foreach (var newThreadState in _newThreadStates)
+                {
+                    DiscardThreadLogState(newThreadState);
+                }
+
+                Thread.Sleep(timeout);
```
Comment (ayende) [inline src/Sparrow/Logging/LoggingSource.cs:708]: Wouldn't the log fill up while we are sleeping?

Comment (grisha-kotler) [reply src/Sparrow/Logging/LoggingSource.cs:708]: The logs are disabled at this point:
https://github.com/ravendb/ravendb/pull/9817/files#diff-cebe7d3f975916f1be4ab533d581c22aR695

Comment (ayende) [reply src/Sparrow/Logging/LoggingSource.cs:708]: Okay

File: src/Sparrow/Logging/LoggingSource.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Sparrow/Logging/LoggingSource.cs]: done

**Review by ayende — APPROVED** [review]

**Review by ppekrol — APPROVED** [review]

**Pull request merged**